python/cudf/cudf/core/dataframe.py

```diff
@@ -300,12 +300,7 @@ def _getitem_tuple_arg(self, arg):
 
         # Step 3: Gather index
         if df.shape[0] == 1:  # we have a single row
```

Comment:
Pretty sure this entire block can be dropped. Both `DataFrame._slice` (since probably #9558) and `DataFrame._apply_boolean_mask` (since probably #9832) appear to maintain the index correctly:

```python
import cudf
from cudf.core.column.column import as_column

df = cudf.DataFrame({"a": [10, 20, 30], "b": [1, 2, 3]}).set_index("a")

print(df._slice(slice(1, 2, 1))):
#     b
# a    
# 20  2

print(df._apply_boolean_mask(as_column([False, True, False]))
#     b
# a    
# 20  2
```

Reply:
Done! Thanks @wence